org.opentosca.planbuilder.type.plugin.ubuntuvm/src/org/opentosca/planbuilder/type/plugin/ubuntuvm/handler/AbstractUbuntuVmHandler.java

```diff
@@ -0,0 +1,241 @@
+package org.opentosca.planbuilder.type.plugin.ubuntuvm.handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.namespace.QName;
+
+import org.opentosca.container.core.tosca.convention.Types;
+import org.opentosca.planbuilder.model.tosca.AbstractNodeTemplate;
+import org.opentosca.planbuilder.model.tosca.AbstractRelationshipTemplate;
+import org.opentosca.planbuilder.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractUbuntuVmHandler implements UbuntuVmHandler {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AbstractUbuntuVmHandler.class);
+
+	abstract protected boolean handleWithLocalCloudProviderInterface(AbstractNodeTemplate abstractNodeTemplate);
+
+	abstract protected boolean handleWithCloudProviderInterface(AbstractNodeTemplate abstractNodeTemplate);
+
+	abstract protected boolean handleGeneric(AbstractNodeTemplate abstractNodeTemplate);
+
+	abstract protected boolean handleWithDockerEngineInterface(AbstractNodeTemplate nodeTemplate);
+
+	@Override
+	public boolean handle(AbstractNodeTemplate nodeTemplate) {
+		if (nodeTemplate == null) {
+			return false;
+		}
+
+		LOG.debug("Checking if nodeTemplate " + nodeTemplate.getId() + " can be handled");
+
+		// cloudprovider node is handled by doing nothing
+		if (org.opentosca.container.core.tosca.convention.Utils
```

Comment:
this full-qualification could be averted by using more expressive names than `Utils`